tests/lib/Microsoft.Fx.Portability.MetadataReader.Tests/TestAssembly.cs

```diff
@@ -124,5 +121,74 @@ public ResourceStreamAssemblyFile(string fileName)
 
             public Stream OpenRead() => s_assembly.GetManifestResourceStream(Name);
         }
+
+        private class ILStreamAssemblyFile : IAssemblyFile
+        {
+            private static readonly string s_ilAsmPath = Path.Combine(Path.GetDirectoryName(s_assembly.Location), "ilasm.exe");
+
+            private readonly ITestOutputHelper _output;
+
+            public ILStreamAssemblyFile(string fileName, ITestOutputHelper output)
+            {
+                _output = output;
+
+                Name = s_assembly.GetManifestResourceNames().Single(n => n.EndsWith(fileName, StringComparison.Ordinal));
+                Exists = Name != null;
+            }
+
+            public string Name { get; }
+
+            public string Version { get; }
+
+            public bool Exists { get; }
+
+            public Stream OpenRead()
+            {
+                if (!File.Exists(s_ilAsmPath))
+                {
+                    throw new FileNotFoundException("Could not find ilasm");
+                }
+
+                var tmp = Path.GetTempFileName();
+
+                using (var fs = File.OpenWrite(tmp))
+                using (var stream = s_assembly.GetManifestResourceStream(Name))
+                {
+                    stream.CopyTo(fs);
+                }
+
+                var psi = new ProcessStartInfo
+                {
+                    Arguments = $"{tmp} /dll",
+                    FileName = s_ilAsmPath,
+                    RedirectStandardOutput = true,
+                    RedirectStandardError = true,
+                    UseShellExecute = false
+                };
+
+                using (var process = Process.Start(psi))
+                {
+                    process.WaitForExit();
+
+                    var stdout = process.StandardOutput.ReadToEnd();
+                    var stderr = process.StandardError.ReadToEnd();
+
+                    _output.WriteLine("ilasm stdout:");
+                    _output.WriteLine(stdout);
+
+                    _output.WriteLine("ilasm stderr:");
+                    _output.WriteLine(stderr);
+
+                    Assert.Equal(0, process.ExitCode);
+                }
+
+                File.Delete(tmp);
+                var output = Path.ChangeExtension(tmp, ".dll");
```

Comment:
The name `output` is a little confusing with `_output`; maybe rename this to outputAssembly or something like that so they're not too similar?